### PR TITLE
MIDI/ULTIMA8: Add XMIDI sequence branch indexes

### PR DESCRIPTION
**Pull request by NMIError**
This adds support for the "sequence branch index" feature of the XMIDI format. 
These "branches" (positions in the MIDI bytestream) are defined in the track 
header. I've added two generic functions to the MidiParser interface to set the 
current parsing position to one of these positions, or to check for the 
existence of a position with a certain number.

Additionally, I've added an option to the MIDI parser to prevent the parser 
from starting playback immediately after parsing, so that a starting position 
(or a track) can be selected first.

Ultima 8 uses sequence branches to mark different playback starting points in 
the music files. I've added support for starting playback at these points.

### CODE REVIEW

**Comment by mduggan** [comment]
This looks great to me, and it seems to be totally backwards-compatible for other engines, but to be honest I'm not super familiar with the midi code so I'll leave it open a couple more days to see if anyone else has comments before merging.

**Comment by OMGPizzaGuy** [comment]
Ran this locally and it works well. 

Starting a combat run consistently gives the following warnings but continues to operate without issue:
WARNING: MidiParser-XMIDI: jumpToIndex called with invalid sequence branch index 0!
WARNING: play() invalid branch index 0!

**Comment by NMIError** [comment]
Thank you for your feedback.

I could not reproduce the issue, but I did see that the code did not properly handle tracks without sequence branches, resulting in the warnings you saw. It did fall back to playing the track from the beginning, so it did work as intended.

I've added an explicit check to see if the track does not have any branches, and then start from the beginning of the track. This should get rid of the warnings.

**Comment by OMGPizzaGuy** [comment]
Makes sense and fixes the issue. Thanks!

File: audio/midiparser.h [hunk 1]
```diff
@@ -377,7 +381,14 @@ class MidiParser {
...
 		  */
-		 mpDisableAllNotesOffMidiEvents = 6
+		 mpDisableAllNotesOffMidiEvents = 6,
+
+		 /**
+		  * Does not automatically start playback after parsing MIDI data
+		  * or setting the track. Use startPlaying to start playback.
+		  * Note that not every parser implementation might support this.
+		  */
+		  mpDoNotAutoStartPlayback = 7
```
Comment (OMGPizzaGuy) [inline audio/midiparser.h:391]: How would you feel about mpDisableAutoStartPlayback name for consistency? Too much of a nitpick?

Comment (NMIError) [reply audio/midiparser.h]: Yes, that's a better name.

**Comment by OMGPizzaGuy** [comment]
Well, this PR has been open for a few weeks and looks like it shouldn't have effects outside engines that plan to use the feature. Merging!

**Comment by NMIError** [comment]
Thanks!

**Pull request merged**